### PR TITLE
Rendering of related_names links in Show page

### LINKED ISSUE #443: Subject/Name facet searches don't work.
See links here: https://cicognara.org/catalog/2475

### PR DESCRIPTION
**Pull request by hectorcorrea**
Indexes names from the Version information into the appropriate Solr documents for Books. This makes sure that the names displayed (from the Version data) in the Show page are found in Solr when clicking on a name in the Show page and issuing a facet search by that value. 

Closes #443 

Below are two examples on how this looks to a user.

## Example 1
This is an example of a value that only exists in the Version data ("Remondini (Firm), printer") and that was failing before when searching for it. The search now finds the correct record since we are indexing the value from the Version data and attaching it to the correct book.

sample1

## Example 2
Another benefit from this PR is that now we are indexing the "Fondo Cicognara (Biblioteca apostolica vaticana)" from the Version data that seems to appear in a lot of the records. A search for this value returns the correct data. 

sample2

### CODE REVIEW

**Comment by hectorcorrea** [comment]
@escowles @kelynch I will be interested in your thoughts on the solution proposed on this PR. The only issue that I believe could cause confusion to users is that *sometimes* we render the "related names" as links and sometimes as text (i.e. not links.)

As indicated in the PR we render them as text (i.e. not links) when the values are not indexed in Solr, but this might be confusing to users. Yet, I could not come up with a better approach.

We could always render them as text (never as links) if we wanted to prevent confusion but then we lose some functionality since sometimes the data in the Book is just as good as the data in the Version record.

Another approach would be to index the data in the Version records but that would make this much more complicate problem.

**Comment by escowles** [comment]
@hectorcorrea It makes sense there isn't always overlap, because of the different data sources. I do think it would be somewhat strange to have items sometimes be links and sometimes not.

I like the idea of indexing the values from the Version records, since that would let us make them always be links, and would probably improve searching functionality in general. But I'm not sure what the complexity (and performance) impact of that would be. Is the indexer already loading those records, or would indexing these fields require a lot more data loading/parsing/etc.?

**Comment by hectorcorrea** [comment]
@escowles The Version information is stored in the Version table but not indexed. 

We could try to update in place the Solr documents (i.e. only the `related_names_display` field) for the Books documents with the information on the Version records as they are processed. I don't think the performance will be an issue, I am mostly concerned on how long it will take me to figure it out since I am not very familiar with the code base. I'll take a stab at it and report back tomorrow.

File: solr/config/solrconfig.xml [hunk 1]
```diff
@@ -407,4 +407,9 @@
     </arr>
   </requestHandler>
 
+  <updateHandler class="solr.DirectUpdateHandler2">
+    <updateLog>
+      <str name="dir">${solr.core0.data.dir:}</str>
+    </updateLog>
+  </updateHandler>
```
Comment (hectorcorrea) [inline solr/config/solrconfig.xml:414]: This change is needed for the atomic updates to work. This setting is [already enabled in our production/staging configuration](https://github.com/pulibrary/pul_solr/blob/main/solr_configs/cicognara/conf/solrconfig.xml#L410-L414), but it was missing in the local development copy.

**Comment by hectorcorrea** [comment]
@escowles @kelynch I updated the indexing process to attach to the Books' Solr documents the name information found in the Version data. This makes sure the searches work as expected. This required usinf Solr's [atomic update](https://solr.apache.org/guide/8_9/updating-parts-of-documents.html#atomic-updates) on the Book records but luckily our Solr and RSolr support it.

**Comment by hectorcorrea** [comment]
Links to records in staging: 
* https://cicognara-staging.princeton.edu/catalog/272
* https://cicognara-staging.princeton.edu/catalog/566

**Review by kelynch — APPROVED** [review]

**Pull request merged**